app/models/concerns/framework_assessmentable.rb

```diff
@@ -65,17 +66,18 @@ def import_nomis_mappings!
     FrameworkNomisMappings::Importer.new(assessmentable: self).call
   end
 
-  def section_progress
-    sections_to_responded.group_by(&:section).map do |section, responses|
+  def calculate_section_progress(responses: required_responses)
```

Comment:
Allow optionally passing in responses, to avoid having to calculate required responses on creation of assessments, which if invoked will load framework questions as well. This is because responses are empty and we don't require to do any calculations on creation.